lib/home/surveys/survey_questions.dart

```diff
@@ -98,33 +135,48 @@ sealed class MultipleChoice extends SurveyQuestion {
   /// if [canType] is set to `true`.
   final List<String> choices;
 
+  int get totalChoices => choices.length + (canType ? 1 : 0);
+
   /// If true, an extra option will appear at the bottom of the list,
   /// allowing the user to type their own response.
   final bool canType;
 
   int? get typingIndex => canType ? choices.length : null;
+
+  @override
+  Map<String, dynamic> get json {
+    final type = switch (this) {
+      RadioQuestion() => 'radio',
+      CheckboxQuestion() => 'checkbox',
+    };
+
+    return {
+      ...super.json,
+      'type': '$type multiple choice',
+      'choices': choices,
+      if (canType) 'custom response allowed': true,
+    };
+  }
```

Comment:
This (along with the other `json` getters in this file) was adapted from #22 by @BooooobZhao.